pkg/commatrix/commatrix.go

```diff
@@ -0,0 +1,222 @@
+	}
+	cmds.AddCommand(NewCmdCommatrixGenerate(streams))
+
+	return cmds
+}
+
+func NewcommatrixOp(streams genericiooptions.IOStreams) *CommatrixOptions {
+	return &CommatrixOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+
+		IOStreams: streams,
+	}
+}
+
+// NewCmdAddRoleToUser implements the OpenShift cli add-role-to-user command.
+func NewCmdCommatrixGenerate(streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewcommatrixOp(streams)
+	cmd := &cobra.Command{
+		Use: "generate",
+		Short: "Generate the communication matrix to create an accurate and up-to-date representation of" +
+			"all ingress communication flows within the OpenShift cluster.",
+		Long:    commatrixLong,
+		Example: CommatrixExample,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+	cmd.Flags().StringVar(&o.format, "format", "csv", "Desired format (json,yaml,csv,nft)")
+	cmd.Flags().StringVar(&o.customEntriesPath, "customEntriesPath", "", "Add custom entries from a file to the matrix")
+	cmd.Flags().StringVar(&o.customEntriesFormat, "customEntriesFormat", "", "Set the format of the custom entries file (json,yaml,csv)")
+	cmd.Flags().BoolVar(&o.debug, "debug", false, "Debug logs")
+	return cmd
+}
+
+// Complete initializes the options based on the provided arguments and flags.
+func (o *CommatrixOptions) Complete(cmd *cobra.Command, args []string) error {
+	// Validate the number of arguments
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %v", args)
+	}
+
+	if o.format == "" {
+		o.format = "csv" // Default format
+	}
+
+	if o.customEntriesPath != "" && o.customEntriesFormat == "" {
+		return fmt.Errorf("you must specify the --customEntriesFormat when using --customEntriesPath")
+	}
+
+	return nil
+}
+
+func (o *CommatrixOptions) Validate() error {
+	// Validate format
+	validFormats := map[string]bool{types.FormatCSV: true, types.FormatJSON: true, types.FormatYAML: true, types.FormatNFT: true}
+	if _, valid := validFormats[o.format]; !valid {
+		return fmt.Errorf("invalid format '%s', valid options are: csv, json, yaml, nft", o.format)
+	}
+
+	// Validate custom entries path and format.
+	if o.customEntriesPath != "" {
+		if o.customEntriesFormat == "" {
+			return fmt.Errorf("you must specify the --customEntriesFormat when using --customEntriesPath")
+		}
+
+		validCustomFormats := map[string]bool{types.FormatCSV: true, types.FormatJSON: true, types.FormatYAML: true}
+		if _, valid := validCustomFormats[o.customEntriesFormat]; !valid {
+			return fmt.Errorf("invalid custom entries format '%s', valid options are: csv, json, yaml", o.customEntriesFormat)
+		}
+	}
+
+	return nil
+}
+
+func (o *CommatrixOptions) Run() (err error) {
+	if o.debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	if o.cs == nil {
```

Comment:
Should the client be created once before running?

Reply:
this is did to use it on the suite test
there im using a mock client like here
https://github.com/openshift-kni/commatrix-cli-plugin/pull/2/files#:~:text=g-,.,-It(%22should%20create%20with%20csv